src/main/java/by/itechart/api/util/specification/UserSpecification.java

```diff
@@ -0,0 +1,46 @@
+package by.itechart.api.util.specification;
+
+import by.itechart.api.entity.User;
+import lombok.Data;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.util.ObjectUtils;
+
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Predicate;
+import javax.persistence.criteria.Root;
+import java.util.ArrayList;
+import java.util.List;
+
+@Data
+public class UserSpecification implements Specification<User> {
+    private final String keyword;
+    private boolean isOnlyActiveUsers = false;
```

Comment:
Since primitive boolean by default init with `false` why not make this property final too?

Reply:
Because we use this field to set extra filtering with active users only. From my side I can change implementation something like this:
`private boolean isOnlyActiveUsers;`

Reply:
You can make it `final` and just add manual init to the one-parameter-constructor:
`this.isOnlyActiveUsers  = false;`